### DISCUSSION TITLE
Turn growstuff into a 'Common sense' gardening assistant / rules repository

### DISCUSSION THREAD

**Discussion by CloCkWeRX** [Growstuff/growstuff#2747] [Ideas]
Much of gardening is about some core activities; ie planting, harvesting (which we model), pruning, watering, transplanting, feeding/fertilizing, etc (which we dont).

Some of these will be manual for some time, others can be automated (watering for example)

Right now, we only capture limited data and draw limited insights, which are actually pretty great - duration of planting, expected harvests, etc. It would be tedious to capture a lot of the day to day activities by asking a user to log in and create a new event/activity; but possible to do from other platforms - an automation engine that turns on a tap could write out a "watered" log item for example.

I think we have touched on the idea of crop guides, community sourced; I'd like to go a bit more granular in modelling 'tips'/'prompts' or 'practices'/'reminders'

The use case here is to generate at first human, then machine friendly actions or questions.

Specific user experiences I'm thinking about:
1) Plan a planting: show a user the recommended layouts, locations, other gardens, specific tips

2) Prompts/events during the planting. Have you done step X at day Y? (Bool)
How would you rate the health of your planting this week?
Presenting those through a notification framework; either passive, where no feedback is required - calendar/rss - or proactive (browser notifications, email, or webhooks), where a confirmation is useful.

3) Rules/notifications that fire when conditions are met (weather events - heat. rain, etc will happen SOON or did just happen) - recommend action X.

For example, "You planted X 30 days ago, and there are no waterings or rain events. X requires a HIGH amount of water and may be distressed".
(I watered/its dead/never remind me about this again as valid options)

Right now, I'm looking at platforms like Home Assistant, which has some good models for Sensors, and some automations(rules); but are too generic to be entirely useful.
If growstuff were a community repository for useful rules around crops; and could fire webhooks or make the event timestamps to ask certain questions available; it would let platforms like that handle the notifications, reminders, or other events with physical devices

**Comment by Br3nda** [comment]
I've looked at this. We'd need to spend monthly dollars on a a bigger database to store this kind of data.
